bigbluebutton-html5/imports/api/voice-users/server/methods/muteToggle.js

```diff
@@ -21,16 +23,29 @@ export default function muteToggle(credentials, userId) {
     mutedBy: requesterUserId,
   };
 
-  const { role: requesterRole } = Users.findOne({
+  const user = Users.findOne({
     meetingId,
     userId: requesterUserId,
   });
 
-  const { muted: isUserMuted } = VoiceUsers.findOne({
+  const voiceUser = VoiceUsers.findOne({
     intId: userId,
   });
 
-  if (requesterUserId !== userId && requesterRole === 'MODERATOR' && !ALLOW_MODERATOR_TO_UNMUTE_AUDIO && isUserMuted) return;
+  if (!user || !voiceUser) return;
+
+  const isListenOnly = voiceUser.listenOnly;
```

Comment:
`const { listenOnly, muted } = voiceUser;`
and then you can combine
`if (listenOnly || muted) return;`
and you can skip the `isMuted &&` on line 47